frontend/src/pages/Profile.tsx

```diff
@@ -0,0 +1,118 @@
+import { Descriptions, DescriptionsProps, Space, Typography } from 'antd';
+import { useEffect, useState } from 'react';
+import { fetchUserEmail } from '../services';
+import useAuth from './UserAuth';
+
+const { Title } = Typography;
+
+interface ProfileData {
+  gender: string;
+  occupation: string;
+  favoriteBook: string;
+  description: string;
+  username: string;
+}
+
+function Profile(): JSX.Element {
+  const [profileData, setProfileData] = useState<ProfileData | null>(null);
+  const { username } = useAuth();
+  const [email, setEmail] = useState('');
+
+  useEffect(() => {
+    // Fetch profile data when the component mounts
+    const fetchProfileData = async () => {
+      try {
+        console.log(username);
+        //   const response = await fetch(`http://localhost:3001/getProfileData/${username}`);
+        const response = await fetch(
+          `http://localhost:3001/api/getProfileData/${username}`
+        );
```

Comment:
This works fine locally, but the hard-coded url here will cause issues when we deploy.

```suggestion
        const response = await fetchProfile(username);
```